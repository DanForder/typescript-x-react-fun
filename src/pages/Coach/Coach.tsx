import { useCallback, useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import Layout from "../../components/Layout/Layout";
import { getCoach } from "../../data/api";
import CoachData from "../../types/server/CoachData";
import { formatName } from "../../utils/nameUtils";

type CoachParams = {
  coachId: string;
};

// to see this page component action, run the app "/coaches/1" appended to the URL 🧐
const Coach: React.FC = () => {
  const { coachId } = useParams<CoachParams>();
  const [coach, setCoach] = useState<CoachData | null>();

  useEffect(() => {
    if (!coachId) return;
    (async () => {
      setCoach((await getCoach(parseInt(coachId))) ?? null);
    })();
  }, [coachId]);

  const coachTsx = useCallback(
    (coach: CoachData | null) =>
      coach ? (
        <>
          <p>Name : {formatName(coach?.firstName, coach?.lastName)}</p>
          <p>Coach Id : {coach?.id}</p>
        </>
      ) : (
        <span>Sorry, no coach found with id of {coachId}</span>
      ),
    [coachId]
  );

  return (
    <Layout navigation>
      <h1>Coach</h1>
      {coach !== undefined ? coachTsx(coach) : <span>Loading...</span>}
    </Layout>
  );
};

export default Coach;
