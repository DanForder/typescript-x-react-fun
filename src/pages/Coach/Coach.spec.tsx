import { customRender } from "../../utils/testUtils";
import Coach from "./Coach";

export default describe("Coach Page", () => {
  it("Renders component in loading state", () => {
    const { container } = customRender(<Coach />);
    expect(container).toMatchSnapshot();
  });
});
