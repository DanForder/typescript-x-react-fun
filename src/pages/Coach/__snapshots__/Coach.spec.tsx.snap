// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coach Page Renders component in loading state 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Coach
      </h1>
      <span>
        Loading...
      </span>
    </main>
  </div>
</div>
`;
